File: aesi-eclipse/plugin/src/main/java/com/virtlink/editorservices/eclipse/AesiModule.java
package com.virtlink.editorservices.eclipse;

import com.virtlink.aesi.eclipse.editors.*;
import com.virtlink.aesi.eclipse.structureoutline.AesiContentProvider;
import com.virtlink.aesi.eclipse.structureoutline.AesiOutlinePage;
import com.virtlink.aesi.eclipse.structureoutline.IContentProviderFactory;
import org.eclipse.jface.text.contentassist.IContentAssistProcessor;

import com.google.inject.AbstractModule;
import com.google.inject.Singleton;
import com.google.inject.assistedinject.FactoryModuleBuilder;
import com.virtlink.aesi.eclipse.DocumentManager;
import com.virtlink.aesi.eclipse.ProjectManager;
import com.virtlink.aesi.eclipse.codecompletion.AesiCompletionProcessor;

public class AesiModule extends AbstractModule {

	@Override
	protected void configure() {
		bind(ProjectManager.class).in(Singleton.class);
		bind(DocumentManager.class).in(Singleton.class);
		bindCodeCompletion();
	}
	
	protected void bindCodeCompletion() {
		install(new FactoryModuleBuilder()
				.implement(IContentAssistProcessor.class, AesiCompletionProcessor.class)
				.build(IContentAssistProcessorFactory.class));
		install(new FactoryModuleBuilder()
				.implement(AesiSourceScanner.class, AesiSourceScanner.class)
				.build(ISourceScannerFactory.class));
		install(new FactoryModuleBuilder()
				.implement(AesiContentProvider.class, AesiContentProvider.class)
				.build(IContentProviderFactory.class));
		install(new FactoryModuleBuilder()
				.implement(AesiOutlinePage.class, AesiOutlinePage.class)
				.build(IOutlinePageFactory.class));



		bind(ColorManager.class).toInstance(new ColorManager());
		bind(AesiSourceViewerConfiguration.class);
	}

}

File: aesi-lsp/src/main/kotlin/com/virtlink/editorservices/lsp/AesiModule.kt
package com.virtlink.editorservices.lsp

import com.google.inject.AbstractModule
import com.google.inject.Singleton
import com.google.inject.assistedinject.FactoryModuleBuilder
import com.virtlink.editorservices.ISessionManager
import com.virtlink.editorservices.SessionManager
import com.virtlink.editorservices.lsp.content.FileSystemContentSource
import com.virtlink.editorservices.lsp.server.AesiLanguageServer
import org.eclipse.lsp4j.services.LanguageServer
import com.virtlink.editorservices.lsp.content.RemoteContentSource
import com.virtlink.editorservices.lsp.content.DocumentContentManager


open class AesiModule : AbstractModule() {
    override fun configure() {
        bind(LanguageServer::class.java).to(AesiLanguageServer::class.java).`in`(Singleton::class.java)
        bind(ISessionManager::class.java).to(SessionManager::class.java).`in`(Singleton::class.java)
        bind(DocumentContentManager::class.java).`in`(Singleton::class.java)
        bind(RemoteContentSource::class.java).`in`(Singleton::class.java)
        bind(FileSystemContentSource::class.java).`in`(Singleton::class.java)

        bind(ProjectManager::class.java).`in`(Singleton::class.java)
        install(FactoryModuleBuilder()
                .implement(DocumentManager::class.java, DocumentManager::class.java)
                .build(IDocumentManagerFactory::class.java))
    }
}